File: Zamba/zambawindows2013/Zamba2010/Zamba.WorkFlow.SQLExecution/Zamba.WorkFlow.Execution.ActivityLibrary/Activity/DOSELECT.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Reflection;
using System.Diagnostics;
using Zamba.Core;
//using Zamba.WFUserControl;

namespace Zamba.WFActivity.Xoml
{
    /// <summary>
    /// This activity show a ArrayList of delays
    /// </summary>
    [RuleDescription("QUERY SELECT"), RuleHelp("Ejecuta una consulta")]
    public partial class DOSELECT : ZRule, IResultActivity, IDOSELECT
	{
        private Zamba.WFExecution.PlayDOSELECT play;
        public DOSELECT()
		{
			InitializeComponent();
            play = new Zamba.WFExecution.PlayDOSELECT(this);
		}
            private String _servidor ;
            private String _dbuser;
            private String _dbpassword;
            private String _dbname;
            private Int32 _servertype;
            private String _SQLName;
            private String _SQL;
            private String _HashTable;
            private String _ExecuteType;

            protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
            {
                play.Play(Results);
                return ActivityExecutionStatus.Closed;
            }
            public Hashtable RuleVariablesInterReglas 
            {
                get{return WFRuleParent.VariablesInterReglas; }
            }

        #region Miembros de IDOSELECT

            string IDOSELECT.Dbname
            {
                get
                {
                    return _dbname;
                }
                set
                {
                    _dbname = value;
                }
            }

            string IDOSELECT.Dbpassword
            {
                get
                {
                    return _dbpassword;
                }
                set
                {
                    _dbpassword = value;
                }
            }

            string IDOSELECT.Dbuser
            {
                get
                {
                    return _dbuser;
                }
                set
                {
                    _dbuser = value;
                }
            }

            string IDOSELECT.ExecuteType
            {
                get
                {
                    return _ExecuteType;
                }
                set
                {
                    _ExecuteType = value;
                }
            }

            string IDOSELECT.HashTable
            {
                get
                {
                    return _HashTable;
                }
                set
                {
                    _HashTable = value;
                }
            }

            string IDOSELECT.SQLName
            {
                get
                {
                    return _SQLName;
                }
                set
                {
                    _SQLName = value;
                }
            }

          

            string IDOSELECT.SQL
            {
                get
                {
                    return _SQL;
                }
                set
                {
                    _SQL = value;
                }
            }

            int IDOSELECT.Servertype
            {
                get
                {
                    return _servertype;
                }
                set
                {
                    _servertype = value;
                }
            }

            string IDOSELECT.Servidor
            {
                get
                {
                    return _servidor;
                }
                set
                {
                    _servidor = value;
                }
            }

            #endregion

        public override void SetParams(object[] Params)
        {
            this._SQLName = Params[0].ToString();
            this._SQL = Params[1].ToString();
            this._HashTable = Params[2].ToString();
            //this._servertype = Int32.Parse(Params[3].ToString());
            this._dbuser = Params[4].ToString();
            this._dbpassword = Params[5].ToString();
            this._dbname = Params[6].ToString();
            this._servidor = Params[7].ToString();
            this._ExecuteType = Params[8].ToString();
        }

        public override System.Collections.Generic.List<object> GetParams()
        {
            System.Collections.Generic.List<object> Params = new System.Collections.Generic.List<object>();
            Params.Add(this._SQLName);
            Params.Add(this._SQL);
            Params.Add(this._HashTable);
            Params.Add(this._servertype);
            Params.Add(this._dbuser);
            Params.Add(this._dbpassword);
            Params.Add(this._dbname);
            Params.Add(this._servidor);
            Params.Add(this._ExecuteType);
            return Params;
        }
    }
}

File: Zamba/ZambaWeb2013/zamba2010/Zamba.WFActivity.Regular/Action/DOSELECT.vb
Imports Zamba.Data
Imports Zamba.Core
Imports System.Xml.Serialization

<RuleCategory("Base de Datos"), RuleDescription("Hacer Select SQL"), RuleHelp("Permite realizar una consulta del tipo select en SQL, configurar la base de datos con la que se va a trabajar y devolver su resultado para ser utilizado en otra regla o etapa a travez de un indice"), RuleFeatures(False)> <Serializable()> _
Public Class DOSELECT
    Inherits WFRuleParent
    Implements IDOSELECT
    Private _isLoaded As Boolean
    Private _isFull As Boolean
    Private playRule As Zamba.WFExecution.PlayDOSELECT
    Public Overrides ReadOnly Property IsFull() As Boolean
        Get
            Return _isFull
        End Get
    End Property
    Public Overrides ReadOnly Property IsLoaded() As Boolean
        Get
            Return _isLoaded
        End Get
    End Property
    Public Overrides Sub FullLoad()

    End Sub
    Public Overrides Sub Load()

    End Sub
    Private Sub New()
        MyBase.New(0, "", 0)
    End Sub
    Public Overrides Sub Dispose()

    End Sub
    Private _servidor As String
    Private _dbuser As String
    Private _dbpassword As String
    Private _dbname As String
    Private _servertype As Int32
    Private _Name As String
    Private _SQL As String
    Private _HashTable As String
    Private _ExecuteType As String = "ESCALAR"

    Public Sub New(ByVal Id As Int64, ByVal Name As String, ByVal wfStepId As Int64, ByVal pSQLName As String, ByVal pSql As String, ByVal pHashTable As String, ByVal pServerName As String, ByVal pDBName As String, ByVal pUserName As String, ByVal pPassword As String, ByVal pServerType As Integer, ByVal pExecuteType As String)
        MyBase.New(Id, Name, wfStepId)
        Me._Name = pSQLName
        Me._SQL = pSql
        Me._HashTable = pHashTable
        Me._servertype = pServerType
        Me._dbuser = pUserName
        Me._dbpassword = pPassword
        Me._dbname = pDBName
        Me._servidor = pServerName
        Me._ExecuteType = pExecuteType
        playRule = New Zamba.WFExecution.PlayDOSELECT(Me)
    End Sub

    'Public Overloads Overrides Function Play(ByVal Results As SortedList) As SortedList
    '[Ezequiel] 09/06/2009 - Se elimino el Try-Catch
    Public Overloads Overrides Function Play(ByVal results As System.Collections.Generic.List(Of ITaskResult)) As System.Collections.Generic.List(Of ITaskResult)
        Return playRule.Play(results)
    End Function
    Public Overloads Overrides Function Playweb(ByVal results As System.Collections.Generic.List(Of ITaskResult), ByRef RulePendingEvent As RulePendingEvents, ByRef ExecutionResult As RuleExecutionResult, ByRef Params As Hashtable) As System.Collections.Generic.List(Of ITaskResult)
        Return playRule.Play(results)
    End Function

    Public Property Dbuser() As String Implements IDOSELECT.Dbuser
        Get
            Return _dbuser
        End Get
        Set(ByVal value As String)
            _dbuser = value
        End Set
    End Property
    Public Property Dbname() As String Implements IDOSELECT.Dbname
        Get
            Return _dbname
        End Get
        Set(ByVal value As String)
            _dbname = value
        End Set
    End Property
    Public Property Dbpassword() As String Implements IDOSELECT.Dbpassword
        Get
            Return _dbpassword
        End Get
        Set(ByVal value As String)
            _dbpassword = value
        End Set
    End Property
    Public Property SqlName() As String Implements IDOSELECT.SQLName
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property
    Public Property HashTable() As String Implements IDOSELECT.HashTable
        Get
            Return _HashTable
        End Get
        Set(ByVal value As String)
            _HashTable = value
        End Set
    End Property
    Public Property Servertype() As Int32 Implements IDOSELECT.Servertype
        Get
            Return _servertype
        End Get
        Set(ByVal value As Int32)
            _servertype = value
        End Set
    End Property
    Public Property SQL() As String Implements IDOSELECT.SQL
        Get
            Return _SQL
        End Get
        Set(ByVal value As String)
            _SQL = value
        End Set
    End Property
    Public Property Servidor() As String Implements IDOSELECT.Servidor
        Get
            Return _servidor
        End Get
        Set(ByVal value As String)
            _servidor = value
        End Set
    End Property
    Public Property ExecuteType() As String Implements IDOSELECT.ExecuteType
        Get
            Return _ExecuteType
        End Get
        Set(ByVal value As String)
            _ExecuteType = value
        End Set
    End Property
End Class

'    Public Overrides Function PrivateCheck() As Boolean
'        Try
'            Dim i As Int32
'            '    If Document.DocTypeId = DocTypeId Then
'            Task.Indexs = Results_Factory.GetIndexs(Task, True)
'            For i = 0 To Me.ReplaceIndexs.Count - 1
'                Dim h As Int32
'                For h = 0 To Task.Indexs.Count - 1
'                    If Me.ReplaceIndexs(i) = "<<" & Task.Indexs(h).Name & ">>" Then
'                        Dim IndexData As String = Task.Indexs(h).data
'                        strSelect = strSelect.Replace(Me.ReplaceIndexs(i), IndexData)
'                    End If
'                Next
'            Next
'            'Ver el tema de instanciar la conexion elegida
'            If Me.dbuser <> "" And Me.dbname <> "" And Me.dbpassword <> "" Then
'                Try
'                    Dim server As New server
'                    server.MakeConnection(Me.servertype, Me.servidor, Me.dbname, Me.dbuser, Me.dbpassword)
'                    Dim conex As IConnection
'                    conex = server.Con
'                    Dim Dt As New DataTable
'                    Dt = server.Con.ExecuteDataset(CommandType.Text, strSelect).Tables(0)
'                    Me.HashTable.Add(Ht, Dt)
'                    Return True
'                Catch ex As Exception
'                   zclass.raiseerror(ex)
'                    Return False
'                End Try
'            Else
'                Me.HashTable.Add(Ht, server.Con.ExecuteDataset(CommandType.Text, strSelect))
'                Return True
'            End If
'            '   Else
'            '     Return False
'            '  End If
'        Catch ex As Exception
'           zclass.raiseerror(ex)
'            Return False
'        End Try
'    End Function
'#End Region

'#Region "Propietary"
'    Dim DocTypeName As String
'    Dim DocTypeId As String
'    Dim strSelect As String
'    Dim ReplaceIndexs As New ArrayList
'    Dim servidor, dbuser, dbpassword, dbname As String
'    Dim servertype As Zamba.Servers.Server.DBTYPES
'    Dim Ht As String

'#End Region

'End Class